### DISCUSSION TITLE
Cannot combine horizontal and vertical split in chained command

### DISCUSSION THREAD

**Discussion by thsymons** [i3/i3#6229] [Q&A] [answered]
I can run the following two command chains separately, and they work as expected

bindsym $mod+Shift+i split h, exec gvim
bindsym $mod+Shift+s split v, exec gvim

But I cannot combine them into a single command, such as:

bindsym $mod+Shift+i split h, exec gvim, split v, exec gvim

If I do, the two windows just split vertically, with no horizontal split

I've tried adding a focus, but that doesn't help.

How would I combine these to behave as they do independently ?

**Comment by thsymons** [comment]
Played around with this a bit and I was able to get it working with i3-msg.
Just combining the commands in i3-msg doesn't solve the problem, as it also seems to ignore some focus commands.
But I got it to work by writing a script that executed multiple i3-msg commands, each with only one focus command at the start.
That alone still didn't work, but when I added a 0.5 second sleep between each command then it all worked as expected.

So for the command chain above, I got it working as follows, in a script:

i3-msg "split h, exec gvim"
sleep 0.5
i3-msg "focus right, split v, exec gvim"

**Comment by thsymons** [comment, marked as answer]
OK, turns out my answer above didn't work that well.  It opens gvim OK, but it opens them in the home directory, with all environment variables lost.  So not too useful.

But I was able to get it working by just launching gvim separately, instead of through i3-msg.  And you only need the sleep between consecutive i3-msg commands.  

i3-msg "split h"
gvim
i3-msg "focus right, split v"
gvim

That seems to get all the windows where I want them.

**Reply by kd4e** [reply]
I have a use case where I need multiple qutebrowser windows maybe as many as 12 (it's a large display).
They need to be in fixed locations on the display (each would show a camera feed).
Each instance will have a default url string (most with extra variables).
I need to be able to click on any one of them, or the side bar, to full screen it (then 'esc' or an F-key to return to the multi-window layout).
Is this similar to what you're doing, please?

**Comment by orestisfl** [comment]
You might benefit from layout saving: https://i3wm.org/docs/layout-saving.html

**Reply by kd4e** [reply]
Cool, thanks!
